csrc/ir/interface_nodes.h

```diff
@@ -550,6 +551,10 @@ class NVF_API TensorView : public Val {
     domain_ = td;
   }
 
+  void setDataType(DataType dtype) {
+    dtype_ = dtype;
+  }
+
```

Comment:
We might have a chain of ops that we skip upcasting for. To avoid replaying that whole chain, I am just changing the dtypes of those intermediate tensors. Currently we have a closed collection of ops, so this is safe. If instead we switched to only upcasting for known ops that require it like arithmetic, then we should be careful that differences in input type only affect the output type and not other aspects of an Expr.

Reply:
Alternatively, we can do the replay and mark replacements. That might be cleaner but there is still the possibility in the future for an op to require more care when changing the dtypes of its inputs.